docs/dev/building/Options.rst

```diff
@@ -0,0 +1,152 @@
+.. _build-options:
+
+#############
+Build Options
+#############
+
+.. contents:: Typical Options
+  :local:
+  :depth: 1
+
+There are a variety of other settings which you can find in CMakeCache.txt in
+your build folder or by running ``ccmake`` (or another CMake GUI). Most
+DFHack-specific settings begin with ``BUILD_`` and control which parts of DFHack
+are built.
+
+Typical usage may look like::
+
+    # Plugin development with updated documentation
+    cmake ./ -G Ninja -B builds/debug-info/ -DCMAKE_INSTALL_PREFIX=<path to DF> -DCMAKE_BUILD_TYPE:string=RelWithDebInfo -DBUILD_DOCS:bool=ON -DBUILD_PLUGINS=1
+    # Core DFHack only
+    cmake ../ -G Ninja -DCMAKE_INSTALL_PREFIX=<path to DF> -DCMAKE_BUILD_TYPE:string=RelWithDebInfo -DBUILD_TESTS -DBUILD_DOCS:0 -DBUILD_PLUGINS=0
+
+.. admonition:: Modifying Build Options
+
+    You can typically run new cmake commands from your build directory to turn on/off options.
+    Of course the generator selection is not something you can change, but the rest are.
+
+    Additionally, you can edit the build settings in CMakeCache.txt. You also have cmake's
+    configuration utility ``ccmake``.
+
+Generator
+---------
+For the uninitiated, the generator is what allows cmake to, of course, generate
+visual studio solution & project files, a makefile, or anything else.
+Your selection of generator comes down to preference and availability.
+
+Visual Studio
+=============
+To generate visual studio project files, you'll need to select a particular version of
+visual studio, and match that to your system's generator list viewed with ``cmake --help``
+
+example::
+
+    cmake .. -G "Visual Studio 17 2022"
+
+Ninja
+=====
+The generally preferred build system where available.
+
+example::
+
+    cmake .. -G Ninja
+
+Install Location
+----------------
+This of course uses the default cmake variable.
+
+Variable: ``CMAKE_INSTALL_PREFIX``
+
+Usage::
+
+    cmake .. -DCMAKE_INSTALL_PREFIX=<path to df>
+
+The path to df will of course depend on your system. If the directory exists it is
+recommended to use ``~/.dwarffortress`` to avoid permission troubles.
+
+Build type
+----------
+Release/Debug, this is the type of build you want. This controls what information
```

Comment:
```suggestion
This is the type of build you want. This controls what information
```

(valid types are listed below, and I don't want to confuse people by suggesting `Debug` will work)

Reply:
Actually, I think Debug may actually work. Though perhaps it doesn't do as expected.

Reply:
It doesn't work at all on Windows, from my understanding (different ABI?)